src/MainWindow/main_window.cpp

```diff
@@ -269,11 +269,10 @@ void MainWindow::newDesignCreated(const QString& design) {
 }
 
 void MainWindow::startStopButtonsState() {
-  const bool inProgress = m_taskManager->status() == TaskStatus::InProgress;
-  const bool consoleInProgress = m_console->isRunning();
-  startAction->setEnabled(!inProgress && !consoleInProgress);
+  startAction->setEnabled(m_taskManager->status() != TaskStatus::InProgress &&
+                          !m_console->isRunning());
   // Enable Stop action when there is something to stop
-  stopAction->setEnabled(inProgress || consoleInProgress);
+  stopAction->setEnabled(isRunning());
```

Comment:
Why are we changing the stopAction state here ? Seems like a refactoring, https://github.com/os-fpga/FOEDAG/pull/763/files#diff-9e31125b4e2b6bd449e09a6b478450cb54bf4fb9ddb23baadb06ad5d9a9bb149R315
correct ?

Reply:
We aren't changing it. The logic was just moved to another place. here is the implementation:
![image](https://user-images.githubusercontent.com/109239890/199246573-7b9339ce-3a6c-4f2e-8fd9-194d76a8c38d.png)
we use the same condition to make sure run_project waits till project finishes running. to avoid code duplication the logic was just moved to a separate method.